src/python_testing/TC_EWATERHTR_2_2.py

```diff
@@ -176,202 +195,275 @@ async def test_TC_EWATERHTR_2_2(self):
         asserts.assert_greater(heaterTypes, 0,
                                f"Unexpected HeaterTypes value - expected {heaterTypes} > 0")
         asserts.assert_less_equal(heaterTypes, Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterHeatSourceBitmap.kOther,
                                   f"Unexpected HeaterTypes value - expected {heaterTypes} <= WaterHeaterHeatSourceBitmap.kOther")
 
-        self.step("4")
+        self.step("5")
         await self.send_test_event_trigger_manual_mode_test_event()
 
-        self.step("4a")
+        self.step("5a")
         heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
         asserts.assert_greater(heatDemand, 0)
         asserts.assert_equal(heatDemand & (~heaterTypes), 0, "heatDemand should only be from declared supported types"),
 
-        self.step("5")
+        self.step("6")
         await self.send_test_event_trigger_water_temperature61C_test_event()
 
-        self.step("5a")
+        self.step("6a")
         heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
         asserts.assert_equal(heatDemand, 0)
 
-        self.step("6")
+        self.step("7")
         await self.send_test_event_trigger_water_temperature20C_test_event()
 
-        self.step("6a")
+        self.step("7a")
         heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
         asserts.assert_greater(heatDemand, 0)
         asserts.assert_equal(heatDemand & (~heaterTypes), 0, "heatDemand should only be from declared supported types"),
 
-        self.step("7")
+        self.step("8")
         await self.send_test_event_trigger_off_mode_test_event()
 
-        self.step("7a")
+        self.step("8a")
         heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
         asserts.assert_equal(heatDemand, 0)
 
-        self.step("8")
+        self.step("9")
         await self.send_boost_command(duration=5, one_shot=True)
 
-        self.step("8a")
+        event_data = events_callback.wait_for_event_report(Clusters.WaterHeaterManagement.Events.BoostStarted)
+        asserts.assert_equal(event_data.boostInfo.duration, 5)
+        asserts.assert_equal(event_data.boostInfo.oneShot, True)
+
+        self.step("9a")
         heatDemand = await self.read_whm_attribute_expect_success(attribute="HeatDemand")
         asserts.assert_greater(heatDemand, 0)
         asserts.assert_equal(heatDemand & (~heaterTypes), 0, "heatDemand should only be from declared supported types"),
 
-        self.step("8b")
+        self.step("9b")
         await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kActive)
 
-        self.step("9")
+        self.step("10")
         time.sleep(6)
 
-        self.step("9a")
+        self.step("10a")
+        event_data = events_callback.wait_for_event_report(Clusters.WaterHeaterManagement.Events.BoostEnded)
```

Comment:
Test step 10a above isn't doing this so just need to tweak the test plan and regenerate the TestSteps for that.

Reply:
Resolved